src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -87,6 +85,8 @@
 
     protected ZigBeeKey networkKey;
 
+    private List<ZigBeeNetworkNodeListener> listeners = new CopyOnWriteArrayList<ZigBeeNetworkNodeListener>();
```

Comment:
You might rather want to use a Set instead of a List in order to avoid duplicates.